tests/src/Integration/Engine/AnnotationProcessingTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Engine\AnnotationProcessingTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Engine;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+use Drupal\Core\Session\SessionManagerInterface;
+
+/**
+ * Tests processing of the ContextDefinition annotation.
+ *
+ * @group rules
+ */
+class AnnotationProcessingTest extends RulesIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $this->sessionManager = $this->prophesize(SessionManagerInterface::class);
```

Comment:
The issue here is that I'm loading a set of plugins and testing the annotations as used in those plugins.  If a plugin has an entity:user context, the plugin will invoke APIs that will require a Session Manager mock.  

When constructing the mocks, I was letting Prophecy tell me what methods of what interfaces it required.  It apparently needed SessionMangerInterface available in services container.

I'll check why $this->sessionManager is not declared here, since I'd agree that it needs to be there if I refer to it.  It may not be necessary, however, since I suspect that adding the service to the container will suffice.